### PR TITLE
Replace library/asn1.php & ASNValue with phpseclib

### LINKED ISSUE #8796: Composer error message
After last gut pull I get this error message every time I run the command  `bin/composer.phar install --no-dev`

```
Generating optimized autoload files
Deprecation Notice: Class Text_LanguageDetect_ISO639Test located in ./vendor/pear/text_languagedetect/tests/Text_LanguageDetect_ISO639Test.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class Text_LanguageDetectTest located in ./vendor/pear/text_languagedetect/tests/Text_LanguageDetectTest.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class HTMLPurifier_Language_en_x_test located in ./vendor/ezyang/htmlpurifier/library/HTMLPurifier/Language/classes/en-x-test.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class Friendica\Util\PidFile located in ./src/Util/Pidfile.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthException located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthConsumer located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthToken located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthSignatureMethod located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthSignatureMethod_HMAC_SHA1 located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthSignatureMethod_PLAINTEXT located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthSignatureMethod_RSA_SHA1 located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthRequest located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthServer located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthDataStore located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class OAuthUtil located in ./library/OAuth1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class ASNValue located in ./library/ASNValue.class.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class ASN_BASE located in ./library/asn1.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///patch/to/tze/friendica/installation/htdocs/bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
```

### Friendica version you encountered the problem
1. git 2020.06-rc
2. php7.4
3. MariaDB 5.5.5-10.1.44-MariaDB-0+deb9u1

### PR DESCRIPTION
**Pull request by nupplaphil**
- Replace `library/asn1.php` & removed a lot of unneeded Crypto methods
- Replace `library/ASNValue.class.php`

Therefore I chose a common security library (http://phpseclib.sourceforge.net).

A optional next part could be a rewrite of `Crypto.php` with using all common functions of phpseclib, so we get rid of our custom-scripted security features. AFAIK in terms of security, it's better to rely on well tested standard libraries.

And after the PR, we're more ready for using `Composer 2.0` (see all the composer notices after an install).

### CODE REVIEW

**Review by MrPetovan — COMMENTED** [review]
Have you verified it works the same?

**Comment by nupplaphil** [comment]
That's why it's a draft :-)
Next step is to add tests for both methods to proof it.

**Comment by nupplaphil** [comment]
even better :-) the phpseclib allows us to remove custom security code because of builtin functionality.

I added a diaspora public rsa key, extracted the needed values with the "old" functionality, saved it to files and replaced it with the new version => worked :)

**Comment by MrPetovan** [comment]
Worked, huh?

File: src/Util/Crypto.php [hunk 1]
```diff
@@ -65,183 +65,53 @@ public static function rsaVerify($data, $sig, $key, $alg = 'sha256')
...
 	 * @return string
 	 */
 	public static function meToPem($m, $e)
 	{
-		$der = self::pkcs8Encode($m, $e);
-		$key = self::DerToPem($der, false);
-		return $key;
+		$rsa = new RSA();
+		$rsa->loadKey([
+								  'e' => new BigInteger($e, 256),
```
Comment (MrPetovan) [inline src/Util/Crypto.php]: Indentation is wrong here, it should be just 3 tabs.

File: src/Util/Crypto.php [hunk 2]
```diff
@@ -65,183 +65,53 @@ public static function rsaVerify($data, $sig, $key, $alg = 'sha256')
...
+	 * @return string The PEM output of this key
 	 */
-	private static function pubRsaToMe($key, &$m, &$e)
+	public static function rsaToPem(string $key)
 	{
-		$lines = explode("\n", $key);
-		unset($lines[0]);
-		unset($lines[count($lines)]);
-		$x = base64_decode(implode('', $lines));
+		$publicKey = new RSA();
```
Comment (MrPetovan) [inline src/Util/Crypto.php]: Please rename `$publicKey` to `$RSA`.

File: src/Util/Crypto.php [hunk 3]
```diff
@@ -65,183 +65,53 @@ public static function rsaVerify($data, $sig, $key, $alg = 'sha256')
...
-	 * @throws \Exception
 	 */
-	public static function pemToMe($key, &$m, &$e)
+	public static function pemToMe(string $key, &$modulus, &$exponent)
 	{
-		$lines = explode("\n", $key);
-		unset($lines[0]);
-		unset($lines[count($lines)]);
-		$x = base64_decode(implode('', $lines));
+		$publicKey = new RSA();
```
Comment (MrPetovan) [inline src/Util/Crypto.php]: Please rename `$publicKey` to `$RSA`.

File: tests/src/Util/CryptoTest.php [hunk 4]
```diff
@@ -51,6 +53,62 @@ public function testRandomDigitsRandomInt()
...
+	/**
+	 * @dataProvider dataPEM
+	 */
+	public function testPemToMe(string $key)
+	{
+		Crypto::pemToMe($key, $m, $e);
+
+		$expectedRSA = new RSA();
+		$expectedRSA->loadKey([
+								  'e' => new BigInteger($e, 256),
```
Comment (MrPetovan) [inline tests/src/Util/CryptoTest.php]: Indentation should be 3 tabs here.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Util/Crypto.php [hunk 5]
```diff
@@ -73,10 +73,11 @@ public static function rsaVerify($data, $sig, $key, $alg = 'sha256')
...
 		$rsa = new RSA();
-		$rsa->loadKey([
-								  'e' => new BigInteger($e, 256),
-								  'n' => new BigInteger($m, 256)
-							  ]);
+		$rsa->loadKey(
+			[
+				'e' => new BigInteger($e, 256),
+				'n' => new BigInteger($m, 256)
+			]);
```
Comment (MrPetovan) [inline src/Util/Crypto.php]: Please choose either the extended form
```suggestion
		$rsa->loadKey(
			[
				'e' => new BigInteger($e, 256),
				'n' => new BigInteger($m, 256)
			]
		);
```
Or the condensed form
```suggestion
		$rsa->loadKey([
			'e' => new BigInteger($e, 256),
			'n' => new BigInteger($m, 256)
		]);
```
But not a mix between both.

Same for the other instance.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Util/Crypto.php (hunk 5 above)
Comment (nupplaphil) [reply src/Util/Crypto.php]: hrmpf .. I had to reinstall my IDE .. Seems like there're some settings left to do ;-)

**Review by MrPetovan — APPROVED** [review]

**Comment by MrPetovan** [comment]
One less file in `library`!

**Comment by MrPetovan** [comment]
Related to #4116

**Pull request merged**